haskell-src/Concordium/Types/Queries.hs

```diff
@@ -1028,3 +1028,30 @@ instance ToJSON BakerRewardPeriodInfo where
               "delegatedCapital" .= brpiDelegatedCapital,
               "isFinalizer" .= brpiIsFinalizer
             ]
+
+-- |Input to queries which take an epoch as a parameter.
+data EpochRequest
+    = -- |Query by genesis index and epoch number.
+      SpecifiedEpoch
+        { -- |The genesis index to query at. The query is restricted to this genesis index, and
+          -- will not return results for other indices even if the epoch number is out of bounds.
+          erGenesisIndex :: !GenesisIndex,
+          -- |The epoch number to query at.
+          erEpoch :: !Epoch
+        }
+    | -- |Query the epoch of a specified block.
+      EpochOfBlock
+        { -- |The block whose epoch is to be used for the query.
+          erBlock :: !BlockHashInput
+        }
+
+-- |Details of which baker won the lottery in a given round in consensus version 1.
+data WinningBaker = WinningBaker
```

Comment:
We should probably add ToJSON instance for this so we can display it in the client.

Reply:
Done.